scripts/updateDeployments.ts

```diff
@@ -0,0 +1,136 @@
+import { Address } from 'viem';
+import { writeFileSync } from 'fs';
+
+type SupportedNetworkResponse = {
+    name: string;
+    chainId: number;
+    blockExplorer: string;
+};
+
+type NetworkRegistryResponse = {
+    [key: string]: {
+        contracts: { name: string; address: Address }[];
+        status: 'ACTIVE' | 'DEPRECATED' | 'SCRIPT';
+        version: 'v2' | 'v3';
+    };
+};
+
+type AbiResponse = {
+    abi: any[];
+};
+
+// {contractName: {chainId: address}, ...}
+type ContractRegistry = {
+    [key: string]: {
+        [key: string]: Address;
+    };
+};
+
+const targetContractsV2 = [
+    'Vault',
+    'BalancerRelayer',
+    'BalancerQueries',
+    'WeightedPoolFactory',
+    'ComposableStablePoolFactory',
+];
+
+const targetContractsV3 = [
+    'Vault',
+    'VaultAdmin',
+    'VaultExtension',
+    'Router',
+    'BatchRouter',
+    'CompositeLiquidityRouter',
+    'BufferRouter',
+    'WeightedPoolFactory',
+    'StablePoolFactory',
+    'GyroECLPPoolFactory',
+    'LBPoolFactory',
+];
+
+const targetContracts = [...targetContractsV2, ...targetContractsV3];
+
+export async function updateDeployments() {
+    const res = await fetch(
+        'https://raw.githubusercontent.com/balancer/balancer-deployments/refs/heads/master/addresses/.supported-networks.json',
+    );
+    const data: Record<string, SupportedNetworkResponse> = await res.json();
+
+    const supportedNetworks = Object.entries(data).map(
+        ([name, { chainId }]) => ({ name, chainId }),
+    );
+
+    const balancerV2Contracts: ContractRegistry = {};
+    const balancerV3Contracts: ContractRegistry = {};
+
+    await Promise.all(
+        supportedNetworks.map(async ({ name: networkName, chainId }) => {
+            const url = `https://raw.githubusercontent.com/balancer/balancer-deployments/refs/heads/master/addresses/${networkName}.json`;
+            const res = await fetch(url);
+            const data: NetworkRegistryResponse = await res.json();
+
+            Object.entries(data)
+                .filter(([_, value]) => value.status === 'ACTIVE')
```

Comment:
I feel like it's a bit weird to have all of those addresses not relevant to the SDK being pulled over here.
Can we also filter by the `targetContracts` arrays above? Then we'll make sure we'll only register addresses relevant to the SDK in the repo 🙃